src/python/gem5/components/boards/abstract_board.py

```diff
@@ -29,21 +29,25 @@
     ABCMeta,
     abstractmethod,
 )
+from pathlib import Path
 from typing import (
+    TYPE_CHECKING,
     List,
     Optional,
     Sequence,
     Tuple,
 )
 
-from m5.objects import (
-    AddrRange,
+from m5.objects.ClockDomain import (
     ClockDomain,
-    IOXBar,
-    Port,
     SrcClockDomain,
-    System,
-    VoltageDomain,
+)
+from m5.objects.System import System
+from m5.objects.VoltageDomain import VoltageDomain
+from m5.objects.XBar import IOXBar
+from m5.params import (
+    AddrRange,
+    Port,
 )
```

Comment:
I would rather not make these kinds of changes. It's a bit different from the traditional gem5 style.

Reply:
That's fair, but without them, mypy can't tell that these items are in the objects module (or if it does, it recognizes them as modules rather than classes), the alternative would be to add `# type: ignore` to every import block.

Reply:
This is mostly due to how the type stubs are generated (I think), maybe that could be something to look at more closely?

Reply:
speaking of imports, are relative or absolute imports prefered?